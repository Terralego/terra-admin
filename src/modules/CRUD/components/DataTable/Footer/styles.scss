@import "@blueprintjs/core/src/components/button/_common.scss";

.table-footer {
  padding-left: 0;
  &__display {
    display: none;
  }
  &__results {
    display: none;
  }
  &__form {
    margin-right: auto !important;
  }
  &__label {
    display: flex;
    margin-bottom: 0 !important;
  }
  &__pagination {
    li {
      display: none;
      margin: 0.2rem;
    }
    .previous,
    .next {
      display: inline;
      [aria-disabled="true"] {
        @include pt-button-disabled();
      }
    }
  }
}

@media all and (min-width: 42em) {
  .table-footer {
    &__results {
      display: inline;
    }
  }
}

@media all and (min-width: 66em) {
  .table-footer {
    &__display {
      display: inline;
    }
    &__pagination {
      li {
        display: inline;
      }
    }
  }
}
