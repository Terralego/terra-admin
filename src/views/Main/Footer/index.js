import React from 'react';

export const Footer = () => (
  <footer>
    Footer
  </footer>
);

export default Footer;
