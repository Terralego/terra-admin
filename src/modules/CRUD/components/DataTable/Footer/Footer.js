import React from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import { parseUrl } from 'query-string';

import {
  Alignment,
  Classes,
  Label,
  Navbar,
  NavbarGroup,
  HTMLSelect as Select,
} from '@blueprintjs/core';

import './styles.scss';

const PAGINATION_RANGE_DISPLAYED = 4;

const getPageFromUrl = string => {
  if (string === null) {
    return null;
  }
  const { query: { page } } = parseUrl(string);
  return Number(page);
};

const getCurrentPage = (prevPage, nextPage) => {
  if (prevPage === null && nextPage === null) {
    return null;
  }
  return prevPage ? prevPage + 1 : nextPage - 1;
};

const Footer = ({
  featuresList: { count = 0, previous = null, next = null } = {},
  onPageChange,
  tableFilters: { page, page_size: pageSize },
  t,
}) => {
  const prevPage = getPageFromUrl(previous);
  const nextPage = getPageFromUrl(next);
  const currentPage = getCurrentPage(prevPage, nextPage);

  if (currentPage === null || !count) {
    return null;
  }

  const forcePage = (page || currentPage) - 1;
  const pageCount = Math.ceil(count / pageSize);
  const pageRangeDisplayed = Math.min(pageCount, PAGINATION_RANGE_DISPLAYED);

  return (
    <Navbar className="table-footer">
      <NavbarGroup align={Alignment.CENTER}>
        <div className="table-footer__form">
          <Label className={`table-footer__label ${Classes.INLINE}`}>
            <span className="table-footer__display">{t('CRUD.table.pagination.display')}</span>
            <Select
              className={Classes.INLINE}
              aria-describedby="table-footer__results"
              onChange={({ target: { value } }) => {
                const nextPageSize = Number(value);
                onPageChange({ nextPageSize });
              }}
              options={[10, 20, 50]}
              value={pageSize}
            />
            <span id="table-footer__results" className="table-footer__results"> {t('CRUD.table.pagination.results')}</span>
          </Label>
        </div>
        <ReactPaginate
          pageCount={pageCount}
          pageRangeDisplayed={pageRangeDisplayed}
          marginPagesDisplayed={1}
          pageLinkClassName={Classes.BUTTON}
          breakLinkClassName={Classes.BUTTON}
          previousLinkClassName={Classes.BUTTON}
          nextLinkClassName={Classes.BUTTON}
          activeLinkClassName={Classes.ACTIVE}
          disabledClassName={Classes.DISABLED}
          previousLabel={t('common.pagination.previous')}
          nextLabel={t('common.pagination.next')}
          onPageChange={onPageChange}
          containerClassName="table-footer__pagination"
          forcePage={forcePage}
        />
      </NavbarGroup>
    </Navbar>
  );
};

Footer.propTypes = {
  featuresList: PropTypes.shape({
    count: PropTypes.number,
    previous: PropTypes.string,
    next: PropTypes.string,
  }),
  onPageChange: PropTypes.func,
  tableFilters: PropTypes.shape({
    page: PropTypes.number,
    page_size: PropTypes.number,
  }),
  t: PropTypes.func,
};

Footer.defaultProps = {
  featuresList: {},
  onPageChange: () => {},
  tableFilters: {},
  t: text => text,
};

export default Footer;
