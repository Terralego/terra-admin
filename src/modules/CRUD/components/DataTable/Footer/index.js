import { withTranslation } from 'react-i18next';
import { withTableFilters } from '../../../services/UserSettingsProvider';
import compose from '../../../../../utils/compose';
import Footer from './Footer';

export default compose(
  withTranslation(),
  withTableFilters(),
)(Footer);
