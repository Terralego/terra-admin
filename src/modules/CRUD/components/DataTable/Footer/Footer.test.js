import React from 'react';
import renderer from 'react-test-renderer';

import Footer from './Footer';

jest.mock('react-paginate', () => ({ ...props }) => <div {...props}>Pagination</div>);

jest.mock('query-string', () => ({
  parseUrl: str => ({
    query: {
      page: {
        str,
      },
    },
  }),
}));

jest.mock('@blueprintjs/core', () => ({
  Alignment: {
    CENTER: 'center',
  },
  Classes: {
    INLINE: 'inline',
  },
  Label: ({ children, ...props }) => <span {...props}>{children}</span>,
  Navbar: ({ children, ...props }) => <nav {...props}>{children}</nav>,
  NavbarGroup: ({ children, ...props }) => <div {...props}>{children}</div>,
  HTMLSelect: ({ options, ...props }) => (
    <select {...props}>
      {options.map(item => <option key={item} value={item}>{item}</option>)}
    </select>
  ),
}));

let props;
beforeEach(() => {
  props = {
    t: text => text,
    featuresList: { count: 10, previous: '1', next: '3' },
    onPageChange: () => {},
    tableFilters: {
      page_size: 10,
    },
    querystring: {},
  };
});

describe('Snapshots', () => {
  it('should render correctly', () => {
    const tree = renderer.create((
      <Footer
        {...props}
      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render anything', () => {
    const tree = renderer.create((
      <Footer
        {...props}
        featuresList={{ count: 10, previous: null, next: null }}
      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
