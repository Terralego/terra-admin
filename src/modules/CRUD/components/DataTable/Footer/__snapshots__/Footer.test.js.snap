// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should not render anything 1`] = `null`;

exports[`Snapshots should render correctly 1`] = `
<nav
  className="table-footer"
>
  <div
    align="center"
  >
    <div
      className="table-footer__form"
    >
      <span
        className="table-footer__label inline"
      >
        <span
          className="table-footer__display"
        >
          CRUD.table.pagination.display
        </span>
        <select
          aria-describedby="table-footer__results"
          className="inline"
          onChange={[Function]}
          value={10}
        >
          <option
            value={10}
          >
            10
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={50}
          >
            50
          </option>
        </select>
        <span
          className="table-footer__results"
          id="table-footer__results"
        >
           
          CRUD.table.pagination.results
        </span>
      </span>
    </div>
    <div
      containerClassName="table-footer__pagination"
      forcePage={NaN}
      marginPagesDisplayed={1}
      nextLabel="common.pagination.next"
      onPageChange={[Function]}
      pageCount={1}
      pageRangeDisplayed={1}
      previousLabel="common.pagination.previous"
    >
      Pagination
    </div>
  </div>
</nav>
`;
